### PR TITLE
New Mongoose 4.0 and Elasticsearch 4.0 version

### PR DESCRIPTION
**Pull request by chetverikov**

### CODE REVIEW

File: test/serialize-test.js [hunk 1]
```diff
@@ -55,10 +55,10 @@ describe('serialize', function(){
       serialized.name.first.should.eql('Jeffrey');
       serialized.name.last.should.eql('Lebowski');
     });
-    it('should serialize object ids as strings', function(){
+/*    it('should serialize object ids as strings', function(){
```
Comment (guumaster) [inline test/serialize-test.js]: why this tests is not useful anymore?

File: test/index-test.js [hunk 2]
```diff
@@ -187,9 +187,12 @@ describe('indexing', function(){
       });
 
       tweet.save(function(){
-        tweet.remove();
+        setTimeout(function(){
```
Comment (guumaster) [inline test/index-test.js]: why is this setTimeout needed? this test is passing ok in master branch.

**Comment by guumaster** [comment]
- Why the 3.0.0 bump? Is there any new functionality incompatible with 2.x? 
- regarding the code style changes. I also don't like the missing semicolons and the comma first, but is the code style already present for the project, could you remove those changes? Is not a good idea to change style on every pull requests.

File: test/serialize-test.js (hunk 1 above)
Comment (chetverikov) [inline test/serialize-test.js]: This test is broken... In mongoose 4.0.0 changed the behavior of manual field populated:
https://github.com/Automattic/mongoose/issues/2678
https://github.com/Automattic/mongoose/issues/2370
https://github.com/Automattic/mongoose/issues/2679
Before after saving in the field was ObjectId, now the value remains the same.

File: test/index-test.js (hunk 2 above)
Comment (chetverikov) [inline test/index-test.js]: OOopps +) My vagrant box is running slow, I had to make the test pass.

**Comment by chetverikov** [comment]
* The version number can not be changed, but it will not be good, because the basic dependencies are big changes.
* Unfortunately, in the commit with the changes of style there are other changes, will not be easy to remove them now +(

File: test/index-test.js (hunk 2 above)
Comment (guumaster) [inline test/index-test.js]: Ok, I've also suffered the "not-real-time" pain of indexing.

**Comment by guumaster** [comment]
Fine with the bump, i see that both dependencies has mayor bumps also.  But until we add .jshintrc or .eslintrc to make a standard coding style I prefer not to merge it. I'll check how to have it checked by Travis also.

What do you think of [AirBnB style](https://github.com/airbnb/javascript/blob/master/README.md)

**Comment by chetverikov** [comment]
Ok.

I also use AirBnB style but with changes. Here is an example of my .jscsrc file -> https://gist.github.com/chetverikov/e3fdbf266c42fcb7dcfd 
and my .jshintrc https://gist.github.com/chetverikov/8c9e6bbb92d1325b83f1

I can be a collaborator +)

**Comment by guumaster** [comment]
I've added jshint, gulp and jscs, check #60.

**Comment by chetverikov** [comment]
Ok.
Why did you cancel my PR? This is not good +(

**Comment by chetverikov** [comment]
I don't understand why you canceled my PR with versions?

**Comment by guumaster** [comment]
Just wanted to separate the style changes from the dependency commits. sorry about that

**Comment by guumaster** [comment]
I've included almost all of your changes. Could you merge latest version 2.1.3?
What is your  opinion on keeping two versions? 
* mongoosastic 2.x => mongoose 3.8.x 
* mongoosastic 3.x => mongoose 4.x

**Pull request closed without merging**